src/Arcus.Security.Providers.AzureKeyVault/Authentication/Interfaces/IKeyVaultAuthentication.cs

```diff
@@ -1,17 +1,17 @@
 ﻿using System.Threading.Tasks;
 using Microsoft.Azure.KeyVault;
 
-namespace Arcus.Security.Providers.AzureKeyVault.Authentication.Interfaces
+namespace Arcus.Security.Providers.AzureKeyVault.Authentication.Interfaces 
 {
     /// <summary>
     ///     Authentication provider for Azure Key Vault
     /// </summary>
-    public interface IKeyVaultAuthenticator
+    public interface IKeyVaultAuthentication
     {
         /// <summary>
         ///     Authenticates with Azure Key Vault
         /// </summary>
-        /// <returns>A <see cref="KeyVaultClient" /> client to use for interaction with the vault</returns>
-        Task<KeyVaultClient> Authenticate();
+        /// <returns>A <see cref="IKeyVaultClient" /> client to use for interaction with the vault</returns>
+        Task<IKeyVaultClient> AuthenticateClient();
```

Comment:
Why don't we just call it Authenticate like it was before? :)
```suggestion
        Task<IKeyVaultClient> Authenticate();
```

Reply:
You should still be able to use IKeyVaultAuthentication.Authenticate if it's for implementing this interface

Reply:
Because it would conflict with the obsolete interface

Reply:
I don't see how. You can implement them both with InterfaceX.Authenticate & InterfaceY.Authenticate

Reply:
Yes, I know that you can get this working with explicit interface implementation, but that also means that you have to cast the instance to the correct interface if you want to call a specific `Authenticate`.
Which means that you could get scenario's where you aren't exactly fully compatible and would get errors while compiling.

Reply:
Given we only call it internally based on the interfaces I think we will not have issues, what am I missing? :)

Reply:
Ah, OK. If you're sure that we only use this internally.